File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        l = 0
        r = len(people)-1
        total = 0
        while r>=l:
            if people[r]+people[l]<=limit:
                total+=1
                r-=1
                l+=1
            elif people[r]+people[l]>limit:
                total+=1
                r-=1
        return total

File: 917-boats-to-save-people/boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
     Arrays.sort(people);
     int r = people.length-1;
     int l =0;
     int trip = 0;
     while(l<=r){
        if(people[l]+people[r]<=limit){
            trip++;
            l++;
            r--;
        }else if(people[l]+people[r]>=limit){
            trip++;
            r--;
        }
     }
     return trip;   
    }
}